cocotb/share/lib/fli/FliObjHdl.cpp

```diff
@@ -217,84 +288,60 @@ int FliEnumObjHdl::set_signal_value(const long value)
         return -1;
     }
 
-    if (m_is_var) {
-        mti_SetVarValue(get_handle<mtiVariableIdT>(), value);
-    } else {
-        mti_SetSignalValue(get_handle<mtiSignalIdT>(), value);
-    }
+    m_fli_intf->mti_set_value(value);
 
     return 0;
 }
 
 int FliLogicObjHdl::initialise(std::string &name, std::string &fq_name)
 {
-    switch (m_fli_type) {
+    mtiTypeIdT _type = m_fli_intf->mti_get_type();
+
+    switch (mti_GetTypeKind(_type)) {
         case MTI_TYPE_ENUM:
             m_num_elems   = 1;
-            m_value_enum  = mti_GetEnumValues(m_val_type);
-            m_num_enum    = mti_TickLength(m_val_type);
+            m_value_enum  = mti_GetEnumValues(_type);
+            m_num_enum    = mti_TickLength(_type);
             break;
         case MTI_TYPE_ARRAY: {
-                mtiTypeIdT elemType = mti_GetArrayElementType(m_val_type);
+                mtiTypeIdT elemType = mti_GetArrayElementType(_type);
 
-                m_range_left  = mti_TickLeft(m_val_type);
-                m_range_right = mti_TickRight(m_val_type);
-                m_num_elems   = mti_TickLength(m_val_type);
+                m_range_left  = mti_TickLeft(_type);
+                m_range_right = mti_TickRight(_type);
+                m_num_elems   = mti_TickLength(_type);
                 m_indexable   = true;
 
                 m_value_enum  = mti_GetEnumValues(elemType);
                 m_num_enum    = mti_TickLength(elemType);
 
-                m_mti_buff    = (char*)malloc(sizeof(*m_mti_buff) * m_num_elems);
-                if (!m_mti_buff) {
-                    LOG_CRITICAL("Unable to alloc mem for value object mti read buffer: ABORTING");
-                    return -1;
-                }
+                m_mti_buff    = new char[m_num_elems];
             }
             break;
         default:
-            LOG_CRITICAL("Object type is not 'logic' for %s (%d)", name.c_str(), m_fli_type);
+            LOG_CRITICAL("Object type is not 'logic' for %s (%d)", name.c_str(), mti_GetTypeKind(_type));
             return -1;
     }
 
     for (mtiInt32T i = 0; i < m_num_enum; i++) {
         m_enum_map[m_value_enum[i][1]] = i;  // enum is of the format 'U' or '0', etc.
     }
 
-    m_val_buff = (char*)malloc(m_num_elems+1);
-    if (!m_val_buff) {
-        LOG_CRITICAL("Unable to alloc mem for value object read buffer: ABORTING");
-    }
+    m_val_buff = new char[m_num_elems+1];
```

Comment:
This is a good change, but the type of change I would expect to see in its own PR

Reply:
I agree, or at least on its own commit.  This change was on a separate commit prior to request to squash.